packages/oasgraph-cli/src/oasgraph.ts

```diff
@@ -30,41 +27,64 @@ program
   .option('--cors', 'enable Cross-origin resource sharing (CORS)')
   .option('--no-viewer', 'do not create GraphQL viewer objects for passing authentication credentials')
   .option('--save <file path>', 'save schema to path and do not start server')
-  .action(function (path) {
-     filePath = path
-  })
   .parse(process.argv)
 
-if (typeof filePath === 'undefined') {
-   console.error('No path provided')
-   console.error('Please refer to the help manual (oasgraph -h) for more information')
-   process.exit(1);
-}
-
+// Select the port on which to host the GraphQL server
+let portNumber: number | string = 3000
 if (program.port) {
   portNumber = program.port
 }
 
-// check if the file exists
-if (fs.existsSync(path.resolve(filePath))) {
-  try {
-    let oas = readFile(path.resolve(filePath))
-    startGraphQLServer(oas, portNumber)
-  } catch (e) {
-    console.error(e)
-  }
+let filePaths = program.args
 
-} else {
-  // falls back to a remote location
-  if (filePath.match(/^https?/g)) {
-    getRemoteFileSpec(filePath).then(remoteContent=> {
-    startGraphQLServer(remoteContent, portNumber)
-  })
-  } else {
-    console.error(`OASGraph reading local file error. File ${filePath} does not exist.`)
-  }
+if (typeof filePaths === 'undefined') {
+  console.error('No path(s) provided')
+  console.error('Please refer to the help manual (oasgraph -h) for more information')
+  process.exit(1)
 }
 
+// Load the OASs based off of the provided paths
+Promise.all(filePaths.map(filePath => {
+  return new Promise((resolve, reject) => {
+    // Check if the file exists
+    if (fs.existsSync(path.resolve(filePath))) {
+      try {
+        resolve(readFile(path.resolve(filePath)))
+      } catch (error) {
+        console.error(error)
+        reject(filePath)
+      }
+
+      return
+
+    // Check if file is in a remote location
+    } else if (filePath.match(/^https?/g)) {
+      getRemoteFileSpec(filePath)
+      .then((remoteContent) => {
+        resolve(remoteContent)
+      })
+      .catch((error) => {
+        console.error(error)
+        reject(filePath)
+      })
+
+      return
```

Comment:
You don't have to return here or in line 58 given the use of the `if ... else` block.

Reply:
Ah right, good call.